src/index.ts

```diff
@@ -1,19 +1,80 @@
 import { Probot } from "probot";
 import fs from 'fs';
+import { V1Secret } from '@kubernetes/client-node/dist/gen/model/v1Secret';
 const k8s = require('@kubernetes/client-node');
+const { createAppAuth } = require("@octokit/auth-app");
+require('dotenv').config();
 
+
+const EXPIRATION_THRESHOLD = 5 * 60000;
+
+// K8s client for using k8s apis.
 const kc = new k8s.KubeConfig();
 kc.loadFromDefault();
 const k8sApi = kc.makeApiClient(k8s.CoreV1Api);
 const k8sContext = kc.getCurrentContext();
-const namespace = (() => {
+const k8snamespace = (() => {
   if (k8sContext === 'inClusterContext') {
     return fs.readFileSync('/var/run/secrets/kubernetes.io/serviceaccount/namespace', 'utf8');
   } else {
     return k8sContext.split('/')[0];
   }
 })();
-console.log(`Operating in the namespace: ${namespace}`);
+console.info(`Operating in the namespace: ${k8snamespace}`);
+
+/**
+* Function that get the new installation token
+* @param    {Number} installId    Installation ID
+* @return   {Array} [Token, ExpiryAt] List containing installation token and expiry time
+*/
+async function getToken(installId: Number) {
+
+  const auth = await createAppAuth({ appId: process.env.APP_ID, privateKey: process.env.PRIVATE_KEY });
+  const appAuth = await auth({ type: "installation", installationId: installId });
+
+  return [appAuth.token, appAuth.expiresAt];
+};
+
+/**
+* Function that patches K8s secret with updated token and expiry time
+* @param    {Number} installId    Installation ID
+* @return   None
+*/
+async function checkExpiryUpdateToken(installId: Number){
+
+  const appSecret = await k8sApi.readNamespacedSecret("peribolos-"+installId, k8snamespace);
+
+  console.info(`Working on the Installation: ${appSecret.body.metadata.name}`);
+  const current_date = new Date();
+  const expiry_date = new Date(appSecret.body.metadata.annotations.expiresAt);
+
+  // check if expiry time with current + constant
+  if (expiry_date.getTime() < (current_date.getTime() + EXPIRATION_THRESHOLD)){
+    console.info(`Token is expired for Installation: ${appSecret.body.metadata.name}`);
+    const resToken = await getToken(installId);
+    var secret: V1Secret = {
+      metadata: {
+        annotations: {
+          "expiresAt": resToken[1]
+        }
+      },
+      stringData: {
+        "token": resToken[0]
+      }
+    }
+    const headers = { 'content-type': 'application/merge-patch+json' };
+    k8sApi.patchNamespacedSecret(appSecret.body.metadata.name, k8snamespace, secret, undefined, undefined, undefined, undefined,{ headers }).then(
+      (response: any) => {
+        console.info(`Successfully update the token in secret ${appSecret.body.metadata.name}: ${response.statusCode}`);
+        console.info();
+      },
+      (err: any) => {
+        console.warn(`Failed to patch secret ${appSecret.body.metadata.name}: ${err}`);
+    });
+  } else {
+    console.info(`Token is not expired for Installation: ${appSecret.body.metadata.name}`);
+  }
```

Comment:
nit, maybe consider reversing this if/else statement since the else branch is one line only -> if you invert the condition you can log and return directly like this:

```js
if (notExpired) {
  console.info("Token is still fresh")
  return
}
// ...continue with the refresh token code
```

I think it can improve readability.

Reply:
sure, i can do it.